docs/src/rules/no-useless-assignment.md

```diff
@@ -0,0 +1,116 @@
+---
+title: no-useless-assignment
+rule_type: suggestion
+related_rules:
+- no-unused-vars
+---
```

Comment:
Let's make sure to include more references for people. See the links in https://github.com/eslint/eslint/issues/17559#issuecomment-1732598182


We can use the "further_reading" key to list these out.